projects/components/facet/facet.service.ts

```diff
@@ -116,6 +118,14 @@ export class FacetService {
         return this.userSettingsService.userSettings["facets"];
     }
 
+    public setAllFacets(facets: FacetState[]) {
+        this.allFacets = facets;
+    }
+
+    public setDefaultFacets(facets: FacetState[]) {
+        this.defaultFacets = facets;
+    }
+
```

Comment:
What is the purpose of these methods? I don't see them being used?

Reply:
These two properties are coming from the Injection in the constructor of the service. Unfortunately, if the facets are not provided in the module, it will always be undefined. Meaning that we have to define the facets in the config.ts file, and cannot use the appService.data property to load the facets from the Sinequa application json. I added these two functions to allow the developer to set these properties from outside the provider. 
Was I wrong in my thinking?